### PR TITLE
External miner interface

### LINKED ISSUE #612: API for external miner
- Replace `/blocks/candidateBlock` with API method that return `m := Blake2b256(hdr.bytesWithoutPow)` and public key.
- Add method, that receives solution (w, nonce, d), construct block with all the data kept inside ErgoMiner and sends it to NodeViewHolder

### PR DESCRIPTION
**Pull request by oskin1**
Close #612

### CODE REVIEW

**Comment by catena2w** [comment]
Switched status to wip. Let's wait until GPU miner implementation and merge after that

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -25,6 +25,12 @@ ergo {
     # Is the node is doing mining
     mining = false
 
+    # Use external miner, native miner is used if set to `false`
+    useExternalMiner = false
```
Comment (catena2w) [inline src/main/resources/application.conf]: Let's have external miner by default

File: src/main/resources/application.conf [hunk 2]
```diff
@@ -25,6 +25,12 @@ ergo {
...
+
+    # Public key mining rewards will be dedicated to
+    miningPubKeyHex = null
```
Comment (catena2w) [inline src/main/resources/application.conf]: Can you do here pubkey or P2PK address (both should work)? I think it's more convenient for users to think in terms of addresses

Comment (oskin1) [reply src/main/resources/application.conf]: Done

**Pull request merged**